src/python/pants/backend/python/targets/python_requirement_library.py

```diff
@@ -17,19 +17,43 @@ class PythonRequirementLibrary(Target):
   :API: public
   """
 
-  def __init__(self, payload=None, requirements=None, **kwargs):
+  def __init__(self, payload=None, requirements=None, prepend_to_pythonpath=False, **kwargs):
     """
+    NB: google-cloud-bigquery==1.10.0 has the following prelude upon import:
+
+        from pkg_resources import get_distribution
+        __version__ == get_distribution("google-cloud-bigquery").version
+
+    which will fail at import time with `pkg_resources.DistributionNotFound`.
+
+    The argument `prepend_to_pythonpath=True` will work around this particular error by ensuring the
+    requirements are within a pex file located at the front of the PEX_PATH when executing
+    pytest. Note that even though this works, it will still show the following warning:
+
+        pex/environment.py:330: PEXWarning: The `pkg_resources` package was loaded from a pex
+        vendored version when declaring namespace packages defined by google-cloud-bigquery
+        1.10.0. The google-cloud-bigquery 1.10.0 distribution should fix its `install_requires` to
+        include `setuptools`.
+
     :param requirements: pip requirements as `python_requirement <#python_requirement>`_\\s.
     :type requirements: List of python_requirement calls
+    :param bool prepend_to_pythonpath: Whether to put this library's requirements at the front of
+                                       the PEX_PATH.
     """
     payload = payload or Payload()
 
     assert_list(requirements, expected_type=PythonRequirement, key_arg='requirements')
+    assert isinstance(prepend_to_pythonpath, bool)
```

Comment:
Note that assertions get stripped away from Python when interpreted with optimizations.

Only use this assertion if you don't care about this being used by developers, otherwise use an exception. Also why use the assertion to begin with? We don't generally do this for type checking.